src/components/RenderRoutes/RenderRoutes.js

```diff
@@ -5,32 +5,25 @@ import { BrowserRouter } from 'react-router-dom';
 
 import Home from '../../pages/Home';
 import Staking from '../../pages/Staking';
-import appStore from '../../store/app.store';
 import Footer from '../layouts/Footer';
-import { MAIN_PAGE, STAKING_PAGE } from '../../utils/constants';
+import { MAIN_PAGE, STAKING_PAGE } from '../../config';
 
 const RenderRoutes = observer(() => (
   <BrowserRouter>
-    {!appStore.auth ? (
-      <Switch>
-        <Route
-          path={MAIN_PAGE}
-          exact
-          render={() => (
-            <>
-              <Home />
-              <Footer />
-            </>
-          )}
-        />
-        <Redirect to={MAIN_PAGE} />
-      </Switch>
-    ) : (
-      <Switch>
-        <Route path={STAKING_PAGE} exact render={() => <Staking />} />
-        <Redirect to={STAKING_PAGE} />
-      </Switch>
-    )}
+    <Switch>
+      <Route
+        path={MAIN_PAGE}
+        exact
+        render={() => (
+          <>
```

Comment:
This render method will cause a problem of resolving `location, history, match` props from react-router.
React router automatic pass this props to router

Can't you just set Footer on every page like:

`...
</Switch>
**<Footer />**
</BrowserRouter>
`

Also, seems like you handle authorization in StakingPage
TBH, I never made an auth in react apps, but react-router docs say it should be handled in routing
https://v5.reactrouter.com/web/example/auth-workflow

@Motzart what do you think about authorization?

Reply:
thank!